plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java

```diff
@@ -305,21 +305,28 @@ static AWSCredentialsProvider buildCredentials(Logger logger, S3ClientSettings c
             }
 
             if (irsaCredentials.getIdentityTokenFile() == null) {
-                return new PrivilegedSTSAssumeRoleSessionCredentialsProvider<>(
-                    securityTokenService,
+                final STSAssumeRoleSessionCredentialsProvider.Builder stsCredentialsProviderBuilder =
```

Comment:
Thanks @reta for the quick fix 👍 

For my understanding, I see the `getCredentials` on `PrivilegedSTSAssumeRoleSessionCredentialsProvider` is already wrapped under SocketAccess and so is `securityTokenService`, how does wrapping the `STSAssumeRoleSessionCredentialsProvider` builder fixes the issue ?

```
        public AWSCredentials getCredentials() {
            return SocketAccess.doPrivileged(credentials::getCredentials);
        }
```

nit: Do we need a unit test to verify the behavior ? I think we don't have tests verifying IRSA creds ? We can take this in separate PR to unblock this. 
https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-s3/build.gradle#L113

Reply:
@dreamer-89 that's the `STSAssumeRoleWithWebIdentitySessionCredentialsProvider.Builder::build()` and `STSAssumeRoleSessionCredentialsProvider.Builder::build()` which were not called in privileged context, the call to `getCredentials()` happens later on. 

We have covered IRSA with test cases and I have not seen this issue and third party integration tests, I think the flow with identity file is vulnerable to security checks, I will try to reproduce it in tests, it is quite very difficult.

Reply:
Thanks @reta for the clarification. I am good with the changes but would wait for one more approval.